eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala

```diff
@@ -101,7 +101,7 @@ class Peer(val nodeParams: NodeParams, remoteNodeId: PublicKey, authenticator: A
       transport ! TransportHandler.Listener(self)
       context watch transport
       val localInit = nodeParams.overrideFeatures.get(remoteNodeId) match {
-        case Some(f) => wire.Init(f)
+        case Some(f) => wire.Init(f, TlvStream(InitTlv.Networks(nodeParams.chainHash :: Nil)))
```

Comment:
We could factor the definition of `TlvStream(InitTlv.Networks(nodeParams.chainHash :: Nil))`.

Actually I think the pattern matching should only return `features` and we should build the `Init` after.